pkg/controller/direct/cloudbuild/workerpool_controller.go

```diff
@@ -197,10 +197,14 @@ func (a *Adapter) Create(ctx context.Context, u *unstructured.Unstructured) erro
 	}
 	status.ObservedState.CreateTime = ToOpenAPIDateTime(created.GetCreateTime())
 	status.ObservedState.UpdateTime = ToOpenAPIDateTime(created.GetUpdateTime())
+	status.ExternalRef = NewResourceRef(created).Get()
 	return setStatus(u, status)
 }
 
 func (a *Adapter) Update(ctx context.Context, u *unstructured.Unstructured) error {
+	if err := a.ValidateExternalResource(); err != nil {
```

Comment:
We do not allow user to update the `spec.resourceID` `spec.projectRef` or `spec.location` to a different value (which means a different GCP object). 

Ideally we can move this to the directbase_controller. (or we already have this check somewhere?)